access-common/src/main/java/edu/unc/lib/dl/ui/util/AnalyticsTrackerUtil.java

```diff
@@ -82,6 +90,28 @@ public void setGaTrackingID(String trackingID) {
         this.gaTrackingID = trackingID;
     }
 
+    /**
+     * Track an event with the specified action for object pid for the active user on the request.
+     *
+     * @param request request
+     * @param action action for the event
+     * @param pid pid identifying the object involved in the event
+     * @param principals
+     */
+    public void trackEvent(HttpServletRequest request, String action, PID pid, AccessGroupSet principals) {
+        try {
+            AnalyticsUserData userData = new AnalyticsUserData(request);
+
+            BriefObjectMetadata briefObject = solrSearchService.getObjectById(new SimpleIdRequest(pid, principals));
+            trackEvent(userData, briefObject.getParentCollection() == null ? "(no collection)"
```

Comment:
i find the method call in lines 106-108 to be a little too compressed for easy readability. would you be willing to break this up into a few simpler steps?